### PR TITLE
expression: fix float64 overflow check in plus/minus real function

### PR DESCRIPTION
**Pull request by wshwsh12**
**Need port to TiKV**

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/23897 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:
We should keep the overflow check logic same with mysql.

How it Works:

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- Keep the overflow check logic same with mysql

### CODE REVIEW

**Review by lzmhhh123 — APPROVED** [review]

File: tools/check/go.mod [hunk 1]
```diff
@@ -20,4 +20,4 @@ require (
 	honnef.co/go/tools v0.0.0-20180920025451-e3ad64cb4ed3
 )
 
-go 1.13
```
[the remark this answers is not in the document]
Comment (wshwsh12) [reply tools/check/go.mod]: Addressed.

**Pull request merged**